src/compiler/checker.ts

```diff
@@ -14175,6 +14215,85 @@ namespace ts {
             }
         }
 
+        function checkSourceFileADefinitionFile(node: Node): boolean {
+            return getSourceFileOfNode(node).fileName.indexOf(".d.ts") > 0;

```

Comment:
Ideally this should check the name ends in ".d.ts", not just contains it somewhere (i.e. `filename.match(/\.d\.ts$/)`)